native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -663,7 +663,13 @@ private void configureAgent(Project project,
                                 FileSystemOperations fileOperations,
                                 Task taskToInstrument,
                                 JavaForkOptions javaForkOptions) {
-        logger.lifecycle("Instrumenting task with the native-image-agent: " + taskToInstrument.getName());
+        //noinspection Convert2Lambda
+        taskToInstrument.doFirst(new Action<Task>() {
```

Comment:
Thank you for fixing this! :) One note - maybe we could log this message only if we're running with the agent enabled - while technically we are instrumenting these tasks every time, it only has an effect if we're running with the agent

Reply:
I'm having troubles figuring out the reliable way to tell if the agent is enabled from this method. Is it possible?

Reply:
Hmm, I think we could capture the `AgentConfiguration` provider from below inside the `doFirst` lambda and check if it's enabled through the `isEnabled`call